include/dsn/utility/metrics.h

```diff
@@ -17,10 +17,12 @@
 
 #pragma once
 
+#include <mutex>
 #include <string>
 #include <unordered_map>
 
 #include <dsn/utility/autoref_ptr.h>
+#include <dsn/utility/singleton.h>
 #include <dsn/utility/ports.h>
```

Comment:
```suggestion
#include <dsn/utility/ports.h>
#include <dsn/utility/singleton.h>
```
Better to keep the headers in lexicographical order